controllers/hetznercluster_controller.go

```diff
@@ -260,6 +238,33 @@ func (r *HetznerClusterReconciler) reconcileNormal(ctx context.Context, clusterS
 	return reconcile.Result{}, nil
 }
 
+func SetControlPlaneEndpoint(hetznerCluster *infrav1.HetznerCluster) (bool) {
+	if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled {
+		if hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 != "<nil>" {
+			defaultHost := hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4
+			defaultPort := int32(hetznerCluster.Spec.ControlPlaneLoadBalancer.Port)
+
+			if hetznerCluster.Spec.ControlPlaneEndpoint == nil {
+				hetznerCluster.Spec.ControlPlaneEndpoint = &clusterv1.APIEndpoint{
```

Comment:
Just noticed with my current code, this line won't take effect, because the objects are passed by value and not by reference hence the Host and Port will never get set right? 
I was thinking maybe I can just pass the hetznerCluster parameter by reference and then use it in the whole function by reference, this way I don't need to return any value either. Is this approach okay?

Reply:
you store everything in hetznerCluster which is a pointer. therefore it should take effect!

Reply:
and yes, the approach is okay so far!

Reply:
Okay so I do not need to alter with the SetControlPlaneEndpoint() function? 
I can begin with the unit test then right?

Reply:
if you didn't change any logic, then all is well! Some minor cosmetics I would change. But go ahead with unit tests for now!

Reply:
I would use the standard Go library for the unit tests of this function, instead of Gingko/gomega we use with envtest

Reply:
Aight I'll get started with the unit tests. Also thank you for making time to review the PR Janis.